h2o-py/h2o/frame.py

```diff
@@ -1011,6 +1011,21 @@ def set_levels(self, levels):
         return H2OFrame._expr(expr=ExprNode("setDomain", self, False, levels), cache=self._ex._cache)
 
 
+    def rename(self, columns=None):
+        """
+        Change names of columns in the frame.
+
+        Dict key is an index or name of the column whose name is to be set.
+        Dict value is the new name of the column.
+
+        :param columns: dict-like transformations to apply to the column names
+        """
+        assert_is_type(columns, None, dict)
```

Comment:
We could additionally check that the list of columns has the expected length

Reply:
`Pandas.rename` approach is to rename 0..n columns and ignore if the column was not found. For example renaming columns `a,b,c,d` with mapping `a->x, f->y` would produce `x,b,c,d` without errors.

Do we want it like that?

`set_names` is for renaming all the columns at once and accepts only full list of columns - no more or less.

Reply:
thanks for explanation!